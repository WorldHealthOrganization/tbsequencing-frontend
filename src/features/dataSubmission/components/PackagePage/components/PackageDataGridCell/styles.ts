import { css } from '@emotion/react';
import appColors from '../../../../../../styles/colors';

const CELL_PADDING_VERTICAL = 15;

export const tableCell = css({
  paddingTop: CELL_PADDING_VERTICAL,
  paddingBottom: CELL_PADDING_VERTICAL,
});

export const error = css({
  backgroundColor: appColors.status.emergencyRedBg,
});
