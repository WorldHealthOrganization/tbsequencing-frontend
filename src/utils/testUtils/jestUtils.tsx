import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
