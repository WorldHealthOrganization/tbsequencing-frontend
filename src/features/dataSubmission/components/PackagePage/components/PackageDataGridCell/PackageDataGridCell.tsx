/** @jsxImportSource @emotion/react */
import React, { ReactNode } from 'react';
import { TableCell } from '@mui/material';
import * as styles from './styles';

interface Props {
  children: ReactNode;
  error?: boolean;
}

export const PackageDataGridCell = ({ children, error }: Props) => {
  const errorStyles = error && styles.error;

  return (
    <TableCell css={[styles.tableCell, errorStyles]}>
      {children}
    </TableCell>
  );
};

export default PackageDataGridCell;
