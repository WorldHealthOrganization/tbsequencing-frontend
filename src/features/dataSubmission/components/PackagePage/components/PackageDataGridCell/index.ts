import PackageDataGridCell from './PackageDataGridCell';

export default PackageDataGridCell;
