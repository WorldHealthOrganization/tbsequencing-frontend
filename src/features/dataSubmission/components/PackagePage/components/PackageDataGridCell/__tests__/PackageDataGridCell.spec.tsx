import React from 'react';
import { screen } from '@testing-library/react';
import { getTreeJSONWithRedux, render } from '../../../../../../../utils/testUtils/jestUtils';
import PackageDataGridCell from '../PackageDataGridCell';

describe('PackageDataGridCell component test', () => {
  const mockContent = 'mockContent';

  test('renders PackageDataGridCell component without error', () => {
    render(
      <PackageDataGridCell>
        {mockContent}
      </PackageDataGridCell>,
    );
  });

  it('Should render children properly', () => {
    render(
      <PackageDataGridCell>
        {mockContent}
      </PackageDataGridCell>,
    );

    screen.getByText(mockContent);
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <PackageDataGridCell>
        {mockContent}
      </PackageDataGridCell>,
    );

    expect(tree).toMatchSnapshot();
  });
});
