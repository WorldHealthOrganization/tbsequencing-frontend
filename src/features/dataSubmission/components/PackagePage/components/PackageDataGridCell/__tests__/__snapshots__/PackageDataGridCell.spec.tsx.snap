// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackageDataGridCell component test Should render correctly 1`] = `
<td
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-sizeMedium css-fdrl5-MuiTableCell-root-PackageDataGridCell"
>
  mockContent
</td>
`;
